lib/galaxy/tools/parameters/validation.py

```diff
@@ -58,7 +58,7 @@ def __init__( self, message, expression ):
         self.expression = expression
 
     def validate( self, value, trans=None ):
-        if re.match( self.expression, value ) is None:
+        if re.match( self.expression, value or '' ) is None:
```

Comment:
I added an issue to track the addition of non-optional text parameters  https://github.com/galaxyproject/galaxy/issues/1867